File: kotlin/src/main/kotlin/exams/exams2/Exam140.kt
package exams.exams2

import kotlin.math.pow

fun findKOr(nums: IntArray, k: Int): Int {
    var digits = IntArray(32) //题目描述为小于31位，这里取整

    for (n in nums) {
        //转换为二进制字符串
        var binStr = n.toString(2)
        for (i in binStr.indices)
            if (binStr[binStr.length - i - 1] == '1')
            //按位统计数位 1
                digits[i]++
    }

    var sum = 0
    for (i in digits.indices) {
        //符合题目的 K-or ，按题目求 乘方 后加入总和
        if (digits[i] >= k) {
            var n = 2.0.pow(i).toInt()
            sum += n
        }
    }

    return sum
}

File: python/src/exams/exams2/Exam140.py
""" 140-找出数组中的 K-or 值

"""

from typing import List


def findKOr(nums: List[int], k: int) -> int:
    digits = [0] * 31

    for n in nums:
        # 转换为二进制字符串
        binStr = format(n, 'b')
        l = len(binStr)
        for i in range(l):
            if binStr[l - i - 1] == '1':
                # 按位统计数位 1
                digits[i] += 1

    sum = 0
    for i in range(len(digits)):
        # 符合题目的 K-or ，按题目求 乘方 后加入总和
        if digits[i] >= k:
            n = 2 ** i
            sum += n

    return sum
